vault/cors.go

```diff
@@ -0,0 +1,112 @@
+package vault
+
+import (
+	"errors"
+	"net/http"
+	"strings"
+)
+
+var errCORSNotConfigured = errors.New("CORS is not configured")
+
+var responseHeaders = map[string]string{
+	"Access-Control-Allow-Headers":     "origin,content-type,cache-control,accept,options,authorization,x-requested-with,x-vault-token",
```

Comment:
Is there any reason to actually have this list? It misses some headers currently valid in Vault. We're also going to have the ability to have selective auditing of headers. It seems like this ought to be `*` unless there is a clear reason why it shouldn't be; if clients pass in random headers we ignore them anyways.